### DISCUSSION TITLE
Open form in modal with inertia route

### DISCUSSION THREAD

**Discussion by jackdaniel9** [inertiajs/inertia#1473] [Help]
Hello, I am new with inertia, vue and laravel. My question is if I am doing the right thing regarding my application.

I have a page with several blocks that the user can modify. When I click on a block's edit button, a modal opens with the block's form.

For the moment, I only add an `<Link:href="route(...)">Edit</Link>` on each block. When the user clicks on the link, the controller returns an "`Internia::render`" of the same page, but with an "`openModal`" parameter of true and the block data.

In my view, I make a condition:
`if(openModal == true) modal.value = true;`

`<TransitionRoot as="template" :show="modal">`

This works great and a modal appears with the block form (component), but is this the right way?

Thank you

**Comment by fabianpnke** [comment]
Hi,
maybe take a look at [https://github.com/lepikhinb/momentum-modal](https://github.com/lepikhinb/momentum-modal) this is a great community package for route-based modals.

**Reply by arthurp93** [reply]
Will it work will ReactJS Inertia?

**Reply by Leonardo8121110010** [reply]
Yes I have seen ReactJs Inertia and Laravel being used

**Comment by vitalijalbu** [comment]
Hi all, i did it without any libraries. But it’s fullwidth. It’s just a page within a modal (using chakra ui)

**Reply by anhvandev** [reply]
@vitalijalbu Hello, may I refer to your method?
